apps/docs/content/guides/functions/examples/amazon-bedrock-image-generator.mdx

```diff
@@ -0,0 +1,150 @@
+---
+id: 'examples-amazon-bedrock-image-generator'
+title: 'Generate Images with Amazon Bedrock'
+description: 'Generate images with Amazon Bedrock and store them in Supabase Storage.'
+tocVideo: 'KIwN2TmkTlg'
+---
+
+[Amazon Bedrock](https://aws.amazon.com/bedrock) is a fully managed service that offers a choice of high-performing foundation models (FMs) from leading AI companies like AI21 Labs, Anthropic, Cohere, Meta, Mistral AI, Stability AI, and Amazon. Each model is accessible through a common API which implements a broad set of features to help build generative AI applications with security, privacy, and responsible AI in mind.
+
+This guide will walk you through an example using the Amazon Bedrock JavaScript SDK in Supabase Edge Functions to generate images using the [Amazon Titan Image Generator G1](https://aws.amazon.com/blogs/machine-learning/use-amazon-titan-models-for-image-generation-editing-and-searching/) model.
+
+## Setup
+
+- In your AWS console, navigate to Amazon Bedrock and under "Request model access", select the Amazon Titan Image Generator G1 model.
+- In your Supabase project, create a `.env` file in the `supabase` directory with the following contents:
+
+```txt
+AWS_DEFAULT_REGION="<youre_region>"
+AWS_ACCESS_KEY_ID="<replace_your_own_credentials>"
+AWS_SECRET_ACCESS_KEY="<replace_your_own_credentials>"
+AWS_SESSION_TOKEN="<replace_your_own_credentials>"
+
+# Mocked config files
+AWS_SHARED_CREDENTIALS_FILE="./aws/credentials"
+AWS_CONFIG_FILE="./aws/config"
+```
+
+### Configure Storage
+
+- [locally] Run `supabase start`
+- Open Studio URL: [locally](http://127.0.0.1:54323/project/default/storage/buckets) | [hosted](https://app.supabase.com/project/_/storage/buckets)
+- Navigate to Storage
+- Click "New bucket"
+- Create a new public bucket called "images"
+
+## Code
+
+Create a new function in your project:
+
+```bash
+supabase functions new amazon-bedrock
+```
+
+And add the code to the `index.ts` file:
+
+```ts index.ts
+// We need to mock the file system for the AWS SDK to work.
+import { prepareVirtualFile } from 'https://deno.land/x/mock_file@v1.1.2/mod.ts'
+
+import { BedrockRuntimeClient, InvokeModelCommand } from 'npm:@aws-sdk/client-bedrock-runtime'
+import { createClient } from 'npm:@supabase/supabase-js'
+import { decode } from 'npm:base64-arraybuffer'
+
+console.log('Hello from Amazon Bedrock!')
+
+Deno.serve(async (req) => {
+  prepareVirtualFile('./aws/config')
+  prepareVirtualFile('./aws/credentials')
+
+  const client = new BedrockRuntimeClient({
+    region: 'us-west-2',
```

Comment:
Do you mean to make this configurable with `AWS_DEFAULT_REGION`? (didn't see it used elsewhere so assumed that was the purpose of that env variable)

Reply:
ah yes, thanks for the catch \o/